core/envelope-sim/src/main/java/fr/sncf/osrd/envelope/Envelope.java

```diff
@@ -184,27 +185,48 @@ public double maxSpeedInRange(double beginPos, double endPos) {
         return maxSpeed;
     }
 
-    /** Computes the time required to get to a given point of the envelope */
-    public long interpolateTotalTimeUS(double position) {
+    private long interpolateUS(double position, boolean isArrivalAt) {
         assert continuous : "interpolating times on a non continuous envelope is a risky business";
         var envelopePartIndex = findLeft(position);
+        if (!isArrivalAt) envelopePartIndex = findRight(position);
         assert envelopePartIndex >= 0 : "Trying to interpolate time outside of the envelope";
         var envelopePart = get(envelopePartIndex);
         return getCumulativeTimeUS(envelopePartIndex) + envelopePart.interpolateTotalTimeUS(position);
     }
 
-    /** Computes the time required to get to a given point of the envelope */
-    public double interpolateTotalTime(double position) {
-        return ((double) interpolateTotalTimeUS(position)) / 1_000_000;
+    @Override
+    public double interpolateArrivalAt(double position) {
+        return ((double) interpolateArrivalAtUS(position)) / 1_000_000;
     }
 
-    /**
-     * Computes the time required to get to a given point of the envelope. The value is clamped to
-     * the [0, envelope length] range.
-     */
-    public double interpolateTotalTimeClamp(double position) {
-        position = Math.min(getEndPos(), Math.max(0, position));
-        return ((double) interpolateTotalTimeUS(position)) / 1_000_000;
+    @Override
+    public long interpolateArrivalAtUS(double position) {
+        return interpolateUS(position, true);
+    }
+
+    @Override
+    public double interpolateDepartureFrom(double position) {
+        return ((double) interpolateDepartureFromUS(position)) / 1_000_000;
+    }
+
+    @Override
+    public long interpolateDepartureFromUS(double position) {
+        return interpolateUS(position, false);
+    }
+
+    @Override
+    public double interpolateArrivalAtClamp(double position) {
+        return ((double) interpolateArrivalAtUS(clamp(position, 0, getEndPos()))) / 1_000_000;
+    }
+
+    @Override
+    public double interpolateDepartureFromClamp(double position) {
+        return ((double) interpolateDepartureFromUS(clamp(position, 0, getEndPos()))) / 1_000_000;
+    }
+
+    /** Returns the time between the departures from two positions of the envelope */
+    public double getTimeBetweenDeparturesFrom(double beginPos, double endPos) {
```

Comment:
Looking at the use cases (time ranges in allowance computations), semantically it should probably be the departure time at the start and arrival time at the end.

But the envelopes don't have stops in this context, so it shouldn't make a difference.

Reply:
As discussed, renamed back to previous name, and commented on the absence of stop in https://github.com/OpenRailAssociation/osrd/pull/7609/commits/b7f044ebd08664f02948f7aafb19982fee469f65